src/Arcus.EventGrid/Contracts/CloudOrEventGridEvent.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Linq;
+using Arcus.EventGrid.Contracts.Interfaces;
+using CloudNative.CloudEvents;
+using GuardNet;
+using Microsoft.Azure.EventGrid.Models;
+
+namespace Arcus.EventGrid.Contracts
+{
+    /// <summary>
+    /// Representation of a supported event in Azure Event Grid.
+    /// This can either be an <see cref="CloudEvent"/> or <see cref="EventGridEvent"/> instance.
+    /// </summary>
+    /// <remarks>
+    ///     This model is not build for custom events.
+    ///     Create your own event by inheriting from <see cref="EventGridEvent{TData}"/>.
+    /// </remarks>
+    public sealed class CloudOrEventGridEvent : IEvent
+    {
+        private readonly CloudEvent _cloudEvent;
+        private readonly EventGridEvent _eventGridEvent;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CloudOrEventGridEvent"/> class.
+        /// </summary>
+        public CloudOrEventGridEvent(CloudEvent cloudEvent)
+        {
```

Comment:
CloudEvents allow you to use different content types in the body but since we use JSON, we should block non-JSON payloads for now and document that.

Reply:
Added guards for this.